core/src/main/java/jenkins/util/TimeDuration.java

```diff
@@ -21,27 +21,59 @@ public TimeDuration(long millis) {
         this.millis = millis;
     }
 
+    /**
+     * Returns the duration of this instance in <em>milliseconds</em>.
+     * @deprecated use {@link #getTimeInMillis()} instead.
+     *
+     * This method has always returned a time in milliseconds, when various callers incorrectly assumed seconds.
+     * And this spread through the codebase. So this has been deprecated for clarity in favour of more explicitly named
+     * methods.
+     */
+    @Deprecated
     public int getTime() {
         return (int)millis;
     }
 
+    /**
+     * Returns the duration of this instance in <em>milliseconds</em>.
+     */
     public long getTimeInMillis() {
         return millis;
     }
 
+    /**
+     * Returns the duration of this instance in <em>milliseconds</em>.
+     */
+    public int getTimeInSeconds() {
+        return (int) (millis / 1000L);
+    }
+
+
     public long as(TimeUnit t) {
         return t.convert(millis,TimeUnit.MILLISECONDS);
     }
 
-    public static @CheckForNull TimeDuration fromString(@CheckForNull String delay) {
-        if (delay==null)
+    /**
+     * Creates a {@link TimeDuration} from the delay passed in parameter
+     * @param delay the delay either in milliseconds without unit, or in seconds if suffixed by sec or secs.
+     * @return the TimeDuration created from the delay expressed as a String.
+     */
```

Comment:
`@since TODO`

Reply:
This method is not new @oleg-nenashev just fixed. Or I'm missing your point?